packages/plugins/src/SearchIndexPlugin.ts

```diff
@@ -73,15 +74,21 @@ const optimizeContentForSearch = async ({
 };
 
 /**
- * Creates a search-data.json file that contains all the information from a page that is required
- * to generate a search index
+ * Creates a search-data.json file that contains all the information
+ * from each page that is required to generate a search index.
  */
 const SearchIndexPlugin: PluginType<Page, SearchIndexPluginOptions> = {
-  async afterUpdate(
+  /**
+   * For each source, parse the data we want to add to the search index and
+   * save it to the global config (to be saved to the filesystem later when
+   * all sources have been loaded).
+   */
+  async $beforeSend(
     mutableFilesystem,
-    { sharedFilesystem, serialiser, ignorePages },
-    { maxLineLength, maxLineCount, keys }
+    { config, serialiser, ignorePages },
+    { disabled, maxLineLength, maxLineCount, keys }
   ) {
+    if (disabled) return;
```

Comment:
Should we filter the plugins and remove disabled plugins rather than have to add a guard to each plugin lifecycle ?

Reply:
Great suggestion. Have added this.